OasisDEX/oasis-borrow
# Improve stability of positions overview page

# [Improve stability of Positions Overview page](https://app.shortcut.com/oazo-apps/story/5060/improve-stability-of-positions-overview-page)
  
## Changes 👷‍♀️
- Create token price services for Coingecko + Coinbase and cache responses from all token price services and serve these results via our existing tokenPrices endpoint.
- Wrap all nested features of the Positions Overview page with their own pipe connected container components so they can all load independently.
- Provide an unobtrusive error fallback message when we fail to resolve a price point for a particular token
  
## How to test 🧪
- Navigate to Positions Overview page and observe how different segments of page load independently
- Observe in the network tab that the api/tokenPrices endpoint is returning data that includes token prices for all service tickers we're interested in
- Run `yarn test`
